### ISSUE TITLE
Modification of Session Expiry Interval in DISCONNECT does not work

### ISSUE THREAD

**Issue by ckrey** [emqx/emqx#1833]
#### Environment

- OS: Debian 9
- Erlang/OTP: 19
- EMQ: 3.0 beta 2

#### Description

CONNECT with a Session Expiry Interval of 3 seconds
DISCONNECT with setting a new Session Expiry Interval of 30 seconds
Wait 10 seconds
Re-CONNECT with CleanStart = false

emqx does not indicate existing session, but should do so

```
19:32:22.471 [debug] MQTT(10.0.2.2:58906): RECV <<16,83,0,4,77,81,84,84,5,2,0,60,5,17,0,0,0,3,0,65,45,91,77,81,84,84,118,53,83,101,115,115,105,111,110,69,120,112,105,114,121,32,116,101,115,116,95,115,69,73,95,115,109,97,108,108,95,100,105,115,99,111,110,110,101,99,116,95,108,97,114,103,101,114,95,114,101,99,111,110,110,101,99,116,93>>
19:32:22.472 [debug] MQTT(@10.0.2.2:58906): RECV CONNECT(Q0, R0, D0, ClientId=-[MQTTv5SessionExpiry test_sEI_small_disconnect_larger_reconnect], ProtoName=MQTT, ProtoVsn=5, CleanStart=true, KeepAlive=60, Username=undefined, Password=undefined)
19:32:22.474 [debug] MQTT(-[MQTTv5SessionExpiry test_sEI_small_disconnect_larger_reconnect]@10.0.2.2:58906): SEND CONNACK(Q0, R0, D0, AckFlags=0, ReasonCode=0)
19:32:22.475 [debug] MQTT(10.0.2.2:58906): SEND <<32,21,0,0,18,39,0,16,0,0,36,2,37,1,42,1,41,1,34,0,0,40,1>>
19:32:22.476 [debug] MQTT(10.0.2.2:58906): Keepalive at the interval of 45
19:32:23.477 [debug] MQTT(10.0.2.2:58906): RECV <<224,7,0,5,17,0,0,0,30>>
19:32:23.477 [debug] MQTT(-[MQTTv5SessionExpiry test_sEI_small_disconnect_larger_reconnect]@10.0.2.2:58906): RECV DISCONNECT(Q0, R0, D0, ReasonCode=0)
19:32:23.478 [debug] MQTT(10.0.2.2:58906): Terminated for normal
19:32:23.479 [info] MQTT(-[MQTTv5SessionExpiry test_sEI_small_disconnect_larger_reconnect]@10.0.2.2:58906): Shutdown for normal
19:32:34.036 [debug] MQTT(10.0.2.2:58908): RECV <<16,78,0,4,77,81,84,84,5,0,0,60,0,0,65,45,91,77,81,84,84,118,53,83,101,115,115,105,111,110,69,120,112,105,114,121,32,116,101,115,116,95,115,69,73,95,115,109,97,108,108,95,100,105,115,99,111,110,110,101,99,116,95,108,97,114,103,101,114,95,114,101,99,111,110,110,101,99,116,93>>
19:32:34.036 [debug] MQTT(@10.0.2.2:58908): RECV CONNECT(Q0, R0, D0, ClientId=-[MQTTv5SessionExpiry test_sEI_small_disconnect_larger_reconnect], ProtoName=MQTT, ProtoVsn=5, CleanStart=false, KeepAlive=60, Username=undefined, Password=undefined)
19:32:34.038 [debug] MQTT(-[MQTTv5SessionExpiry test_sEI_small_disconnect_larger_reconnect]@10.0.2.2:58908): SEND CONNACK(Q0, R0, D0, AckFlags=0, ReasonCode=0)
19:32:34.039 [debug] MQTT(10.0.2.2:58908): SEND <<32,21,0,0,18,39,0,16,0,0,36,2,37,1,42,1,41,1,34,0,0,40,1>>
19:32:34.039 [debug] MQTT(10.0.2.2:58908): Keepalive at the interval of 45
19:32:35.043 [debug] MQTT(10.0.2.2:58908): RECV <<224,0>>
19:32:35.043 [debug] MQTT(-[MQTTv5SessionExpiry test_sEI_small_disconnect_larger_reconnect]@10.0.2.2:58908): RECV DISCONNECT(Q0, R0, D0, ReasonCode=0)
19:32:35.043 [debug] MQTT(10.0.2.2:58908): Terminated for normal
19:32:35.044 [info] MQTT(-[MQTTv5SessionExpiry test_sEI_small_disconnect_larger_reconnect]@10.0.2.2:58908): Shutdown for normal
```

**Comment by Gilbert-Wong** [comment]
It has not been not implemented, same as https://github.com/emqx/emqx/issues/1834

**Comment by tigercl** [comment]
hi, @ckrey, it has been fixed in #1845
